Author: Andreas Beckmann <debian@abeckmann.de>
Author: Antonio Ospite <ospite@studenti.unina.it>
Description: Make the kernel module build against kernels >= 3.4
 In newer kernels asm/system.h is not available anymore, its content has
 been scattered into other header files, to avoid circular dependencies,
 see https://lkml.org/lkml/2012/3/7/389 and
 https://lkml.org/lkml/2012/3/28/381
 .
 All the needed pieces are already brought in by other header files in
 this case, so just avoid including asm/system.h when compiling against
 newer kernels.
 .
 This patch has been tested against 2.6.32, 3.2, 3.3 and 3.4.0-rc1.
Bug-Debian: http://bugs.debian.org/667719

--- a/nv-linux.h
+++ b/nv-linux.h
@@ -113,7 +113,9 @@
 #include <linux/timer.h>
 
 #include <asm/div64.h>              /* do_div()                         */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,4,0)
 #include <asm/system.h>             /* cli, sli, save_flags             */
+#endif
 #include <asm/io.h>                 /* ioremap, virt_to_phys            */
 #include <asm/uaccess.h>            /* access_ok                        */
 #include <asm/page.h>               /* PAGE_OFFSET                      */
